identity-service/sequelize/migrations/20200325195706-unique-user-listens.js

```diff
@@ -0,0 +1,32 @@
+'use strict'
+
+/**
+ * Makes entries in the user list table unique on track id and user id.
+ * Deletes currently conflicting entries. We aren't sure how these arose in the
+ * first place.
+ */
+module.exports = {
+  up: (queryInterface, Sequelize) => {
+    return queryInterface.sequelize.transaction(async (transaction) => {
+      await queryInterface.sequelize.query(`
+        DELETE FROM "UserTrackListens"
+        WHERE "id" in (
+          SELECT "id" FROM (
+            SELECT *,
+            ROW_NUMBER() OVER (PARTITION BY "userId", "trackId" ORDER BY "userId", "trackId") as rowNumber
+            FROM "UserTrackListens"
+          ) A
+          WHERE A.rowNumber > 1
+        )`)
```

Comment:
do you want to add the query to this transaction?

Reply:
Wow turns out you can't actually add a constraint in a transaction

https://sequelize.org/master/class/lib/query-interface.js~QueryInterface.html#instance-method-addConstraint